src/concat.js

```diff
@@ -13,8 +13,8 @@ var invoker = require('./invoker');
  * @category List
  * @sig [a] -> [a] -> [a]
  * @sig String -> String -> String
- * @param {Array|String} a
- * @param {Array|String} b
+ * @param {Array|String|Object} a
+ * @param {Array|String|Object} b
```

Comment:
I believe there is a standard for describing duck-typed objects in JSDoc as `{a: number}`, which would describe an object with a property `a` whose value is a `number`.

So this could be described as `{Array|String|{concat: function}}`.